### PR TITLE
Update onedrive-authorize

### PR DESCRIPTION
**Pull request by mysoniw**
one drive for personal does not need  client secret params

### CODE REVIEW

File: onedrive-authorize [hunk 1]
```diff
@@ -9,7 +9,7 @@ echo "https://login.live.com/oauth20_authorize.srf?client_id=${api_client_id}&sc
 echo ""
 echo "When ready, please enter the value of the code parameter (from the URL of the blank page) and press return"
 read code
-api_json_result=$(curl -s -d "client_id=${api_client_id}&client_secret=${api_client_secret}&code=${code}&grant_type=authorization_code&redirect_uri=https://login.live.com/oauth20_desktop.srf" -X POST "${api_auth_url}")
```
Comment (fkalis) [inline onedrive-authorize]: According to the official documentation, the parameter `client_secret` is required: https://dev.onedrive.com/auth/msa_oauth.htm#step-2-redeem-the-code-for-access-tokens

Even though it works without providing the secret, I would leave it there until Microsoft changes the documentation.

**Comment by weismat** [comment]
This patch seems to be required now. The current master does not work.

**Comment by la-costa** [comment]
I agree with @weismat, this patch works to me.

**Comment by fkalis** [comment]
Microsoft has two types of applications:

microsoft-apps

The master branch supports only the Live SDK applications for OneDrive Personal. You have to make sure, that your App ID contains **no dashes**. If it does, you most likely have created a converged application.

If you have created a converged application and try to authorize your app with the script from the master branch, you receive the error `"Public clients can't send a client secret.`. In this case, you should checkout [this feature branch,](/fkalis/bash-onedrive-upload/tree/feature/28-converged-applications) and follow the instructions from the README file (from that branch).

**Comment by fkalis** [comment]
After some more testing I will merge the [feature branch](/fkalis/bash-onedrive-upload/tree/feature/28-converged-applications) into the master. That means no more support for the old (deprecated?) Live SDK application, but allows the script to use the new Graph API from Microsoft.

**Pull request closed without merging**